scormplayer/scorm-impl/service/src/java/org/sakaiproject/scorm/service/impl/ScormEntityProducer.java

```diff
@@ -0,0 +1,474 @@
+        try {
+            contentHostingService.removeCollection(collectionId);
+        } catch (Exception e) {
+            log.debug("Unable to remove temporary SCORM collection {} during copy cleanup", collectionId, e);
+        }
+    }
+
+    private void purgeContentPackage(ContentPackage pkg) {
+        if (pkg == null) {
+            return;
+        }
+
+        SecurityAdvisor advisor = (userId, function, reference) -> SecurityAdvice.ALLOWED;
+        securityService.pushAdvisor(advisor);
+        try {
+            try {
+                contentPackageDao.remove(pkg);
+            } catch (Exception e) {
+                log.warn("Unable to mark SCORM package {} as deleted: {}", pkg.getContentPackageId(), e.getMessage());
+            }
+
+            try {
+                if (pkg.getResourceId() != null) {
+                    scormResourceService.removeResources(pkg.getResourceId());
+                }
+            } catch (ResourceNotDeletedException e) {
+                log.warn("Unable to remove SCORM resources for {}: {}", pkg.getContentPackageId(), e.getMessage());
+            }
+        } finally {
+            securityService.popAdvisor(advisor);
+        }
+    }
+
+    @Override
+    public void hardDelete(String siteId) {
+        if (siteId == null || siteId.isBlank()) {
+            return;
+        }
+
+        try {
+            List<ContentPackage> packages = scormContentService.getContentPackages(siteId);
+            packages.forEach(this::purgeContentPackage);
+        } catch (ResourceStorageException e) {
+            log.warn("Unable to hard delete SCORM content for {}: {}", siteId, e.getMessage());
+        }
+    }
+
+    private String extractResourceId(String collectionId) {
+        if (collectionId == null) {
+            return null;
+        }
+        String trimmed = collectionId.endsWith(Entity.SEPARATOR) ? collectionId.substring(0, collectionId.length() - 1) : collectionId;
+        if (trimmed.startsWith(ScormConstants.ROOT_DIRECTORY)) {
+            return trimmed.substring(ScormConstants.ROOT_DIRECTORY.length());
+        }
+        return trimmed;
+    }
+
+    private String buildPackageReference(String context, Long contentPackageId) {
+        if (context == null || contentPackageId == null) {
+            return null;
+        }
+        return new StringBuilder(REFERENCE_ROOT)
+                .append(Entity.SEPARATOR)
+                .append(context)
+                .append(Entity.SEPARATOR)
+                .append(contentPackageId)
+                .toString();
+    }
+
+    private String buildResourcePath(String resourceId) {
+        return ScormConstants.ROOT_DIRECTORY + resourceId;
+    }
+
+    private String buildCollectionPath(String resourceId) {
+        return buildResourcePath(resourceId) + Entity.SEPARATOR;
+    }
+
+    private String resolveUserId() {
+        String userId = sessionManager != null ? sessionManager.getCurrentSessionUserId() : null;
```

Comment:
given that this happens in a separate thread the call to sessionManager is probably not reliable or at least needs verification that the session is correctly opened for the thread

Reply:
I went back reviewed that the session is correctly added to the thread